### PR TITLE
KYRA: support hebrew fan translation for Legend of Kyrandia 2: The Hand of Fate

### PR DESCRIPTION
**Pull request by BLooperZ**
Following #2683 
Here is Kyrandia 2 :)

this PR adds support for hebrew fan translation
changes includes:
Add kyra2 hebrew fan translation detection entry
Add kyra2 hebrew strings to KYRA.DAT (+Bump KYRA.DAT version)
Convert strings to be written Right-to-Left in main menu + intro sequence + spellbook

### CODE REVIEW

File: engines/kyra/detection_tables.h [hunk 1]
```diff
@@ -829,6 +829,20 @@ const KYRAGameDescription adGameDescs[] = {
...
+			"CD",
+			AD_ENTRY2s("FATE.PAK", "28cbad1c5bf06b2d3825ae57d760d032", -1,
+					   "8FAT.FNT", "12424362a537e1335b10323c4013bb1d", -1),
+			Common::HE_ISR,
+			Common::kPlatformDOS,
+			ADGF_DROPLANGUAGE | ADGF_CD,
+			GUIO5(GUIO_MIDIADLIB, GUIO_MIDIMT32, GUIO_MIDIGM, GUIO_MIDIPCSPK, GUIO_RENDERVGA)
+		},
+		KYRA2_CD_FLAGS
+	},
```
Comment (athrxx) [inline engines/kyra/detection_tables.h:844]: Does it work properly for you like that? Normally you would use KYRA2_CD_FAN_FLAGS(Common::HE_ISR, Common::EN_ANY) to tell the engine which of the three included languages gets replaced.

**Comment by athrxx** [comment]
Nice :-) 
It looks pretty safe to me (no changes to the non-Hebrew parts).
Please have a look at the detection entry, the flags might need to be changed like I said in the comment.

File: engines/kyra/detection_tables.h (hunk 1 above)
Comment (BLooperZ) [reply engines/kyra/detection_tables.h:844]: It does (after investigating why, seems like _lang is initialized to 0). 
Still updated the flags as recommended, though.
Thanks you.

**Comment by athrxx** [comment]
Thanks. I'll merge it right away so that it gets into the upcoming release...

File: engines/kyra/gui/gui_v1.cpp [hunk 2]
```diff
@@ -599,26 +599,34 @@ void MainMenu::printString(const char *format, int x, int y, int col1, int col2,
 	Common::String string = Common::String::vformat(format, vaList);
 	va_end(vaList);
 
+	Common::String revBuffer;
+	const char *cstr = string.c_str();
+	if (_vm->gameFlags().lang == Common::HE_ISR) {
+		for (int i = string.size() - 1; i >= 0; --i)
+			revBuffer += string[i];
```
Comment (orgads) [inline engines/kyra/gui/gui_v1.cpp:606]: Why don't use you `Common::convertBiDiString`?

Comment (BLooperZ) [reply engines/kyra/gui/gui_v1.cpp:606]: I have tried, but IIRC it had issues with brackets (like in "[ Autosave ]"), so I just opted for reversing the texts, knowing the result will always be the exact same text spelled backwards is very predictable.

**Pull request merged**